src/manifest.c

```diff
@@ -682,12 +695,22 @@ struct manifest *load_mom_err(int version, bool latest, bool mix_exists, int *er
 struct manifest *load_manifest(int current, int version, struct file *file, struct manifest *mom, bool header_only)
 {
 	struct manifest *manifest = NULL;
+	struct file *peer = NULL;
 	int ret = 0;
 	bool retried = false;
 
 retry_load:
 	ret = retrieve_manifests(current, version, file->filename, file, file->is_mix);
 	if (ret != 0) {
+		if (file->is_iterative == 1 && file->full_manifest && !retried) {
+			/* Since the iterative manifest failed to be loaded,
+			 * fall back to using the full manifest */
+			file->full_manifest->peer = file->peer;
+			file = file->full_manifest;
+			file->peer->peer = file;
```

Comment:
Are you sure you don't need to check if file->peer isn't NULL before doing that? (I'm just asking you that because you were checking that before. It looks correct the way it is)

Reply:
Well, the update workflow doesn't really need it because by the time the code gets to this part  file->peer is not going to be NULL since manifests were already linked here https://github.com/clearlinux/swupd-client/pull/565/files#diff-423d35f9c8cae7b481adc9f5a00d0f58R401. So, I was doing that just to be safe. 

But since the update workflow is the only one that uses iterative manifests I guess there is no point on adding that check, that is why I decided to remove it.

Reply:
We are going to need to update this since we plan on *not* linking manifests when delta manifests are available since delta manifests are not signed. In this case file->peer will be null and file->peer->peer will fail.